syntax = "proto3";

package DATA;


message Result {bool success = 1;}
message Data {string msg = 1;}
message StoreReq{string username =1; string password = 2; string msg = 3;}
message Passcode {string code = 1;}
message UserPass{ string username = 1; string password = 2;}

service DB
{
    rpc RegisterUser(UserPass) returns (Result);
    rpc StoreData(StoreReq) returns (Result);
    rpc GenPasscode(UserPass) returns (Passcode);
    rpc GetData(UserPass) returns (Data);
    rpc GetAuthData(Passcode) returns (Data);
}